### PR TITLE
Enable division and bit manipulation extensions in benchmarks

### LINKED ISSUE #613: Use division and bit extensions in IPC benchmarks
Currently the benchmarks are compiled with the C and Zmmul extensions only.

### PR DESCRIPTION
**Pull request by xThaid**
Blocked until the new image is in the package registry.

Fixes #613.

I ran the benchmarks here: https://github.com/xThaid/coreblocks/actions/runs/8239973106/job/22534443637

### CODE REVIEW

**Comment by tilk** [comment]
The source of the riscv-arch-test test failure is actually in the building workflow:
```
 /__w/coreblocks/coreblocks/test/external/riscof/riscv-arch-test/riscv-test-suite/rv32i_m/I/src/jalr-01.S:72: Error: illegal operands `la x0,5b'
```
This looks unexpected.

**Comment by xThaid** [comment]
I updated the toolchain, because I was getting errors when tried to compile the toolchain with Zb* extensions.

File: .github/workflows/benchmark.yml [hunk 1]
```diff
@@ -63,7 +63,7 @@ jobs:
   build-perf-benchmarks:
     name: Build performance benchmarks
     runs-on: ubuntu-latest
-    container: ghcr.io/kuznia-rdzeni/riscv-toolchain:2023.11.19_v
+    container: ghcr.io/xthaid/riscv-toolchain:2024.03.12
```
Comment (xThaid) [inline .github/workflows/benchmark.yml:66]: I will revert it to `ghcr.io/coreblocks` when there is an updated image in our org registry.

Comment (tilk) [reply .github/workflows/benchmark.yml:66]: Done.

**Review by tilk — APPROVED** [review]

**Review by lekcyjna123 — APPROVED** [review]

**Comment by tilk** [comment]
It's pretty nice that the benchmark job now runs in 14 minutes instead of 16-17.

**Comment by piotro888** [comment]
I reported the `jalr` test problem to `riscv-arch-tests` https://github.com/riscv-non-isa/riscv-arch-test/issues/444

cause in binutils: https://github.com/bminor/binutils-gdb/commit/27b33966b18ed8bf1701a60999448224b1d28273

**Pull request merged**